### PR TITLE
[refactor] engine: duckduckgo - https://html.duckduckgo.com/html

### LINKED ISSUE #3927: duckduckgo blocked by CAPTCHA
Not often, but at some point ddg might block the SearXNG requests by a CAPTCHA,

The CAPTCHA goes away when opening duckduckgo.com in a browser / from the SearXNG IP .. as described here: [Answer CAPTCHA from server’s IP](https://docs.searxng.org/admin/answer-captcha.html#answer-captcha-from-server-s-ip)

I have created this ticket with the intention that we (I) re-engineer the ddg engine ... especially the determination of the [vqd value](https://docs.searxng.org/dev/engines/online/duckduckgo.html#searx.engines.duckduckgo.get_vqd) should be paid attention to again, this should actually avoid CAPTCHAs

----

May we consider to switch to https://html.duckduckgo.com/html/ the URL where ddg routes your browser, when JS is disabled.

https://github.com/searxng/searxng/blob/058a072404a4e9ec4c5a0a957fe06ce66e652938/searx/engines/duckduckgo.py#L56-L56

### PR DESCRIPTION
**Pull request by return42**
The entire source code of the duckduckgo engine has been reengineered and purified.

1. DDG used the URL https://html.duckduckgo.com/html for no-JS requests whose response is also easier to parse than the previous https://lite.duckduckgo.com/lite/ URL

2. the bot detection of DDG has so far caused problems and often led to a CAPTCHA, this can be circumvented using `'Sec-Fetch-Mode'] = “navigate”`

Closes: https://github.com/searxng/searxng/issues/3927

### CODE REVIEW

**Comment by return42** [comment]
I test this PR on my instance at https://darmarit.org/searx/stats?engine=duckduckgo .. I will record my observations in this commentary:

--------------

Some bots jump directly to one of the following pages.  There is no vqd value for this and DDG recognizes this request as a request from a bot. This lowers the reputation of the SearXNG IP and DDG starts to activate CAPTCHAs.

--------------

ddg supports queries like `!ddg acidyl site:google.com`. However, a CAPTCHA must be filled in for [advanced search syntax](https://duckduckgo.com/duckduckgo-help-pages/results/syntax/). 

Tested on my instance: [Removing advanced search syntax](https://github.com/searxng/searxng/compare/cbadb12ed13220603fabeafdd6dba6dace346fbd..a8e65e92c4055114f41ca4e8cc6fae74242eecc0) paired with PR:

- https://github.com/searxng/searxng/pull/3965

gives back a reliability of the duckduckgo engine of 100%.

File: searx/engines/duckduckgo.py [hunk 1]
```diff
@@ -309,40 +349,37 @@ def response(resp):
...
-    # In the last <tr> is the form of the 'previous/next page' links
-    tr_rows = tr_rows[:-1]
+        item = {}
+        title = eval_xpath(div_result, './/h2/a')
+        if not title:
+            # this is the "No results." item in the result list
+            continue
+        item["title"] = extract_text(title)
+        item["url"] = eval_xpath(div_result, './/h2/a/@href')[0]
+        item["content"] = extract_text(eval_xpath(div_result, './/a[@class="result__snippet"]')[0])
```
Comment (Bnyro) [inline searx/engines/duckduckgo.py]:
```suggestion
        item["content"] = extract_text(eval_xpath(div_result, './/a[contains(@class, "result__snippet"]'))[0])
```

File: searx/engines/duckduckgo.py [hunk 2]
```diff
@@ -309,40 +349,37 @@ def response(resp):
...
 
-    len_tr_rows = len(tr_rows)
-    offset = 0
+        results.append(item)
 
-    zero_click_info_xpath = '//html/body/form/div/table[2]/tr[2]/td/text()'
+    zero_click_info_xpath = '//div[@id="zero_click_abstract"]'
     zero_click = extract_text(eval_xpath(doc, zero_click_info_xpath)).strip()
 
-    if zero_click and "Your IP address is" not in zero_click and "Your user agent:" not in zero_click:
```
Comment (Bnyro) [inline searx/engines/duckduckgo.py:345]: If I search for `ip` now, it shows me the ip address of the server, which has been avoided with this line. Thus I'd prefer keeping this.

**Comment by Bnyro** [comment]
Works fine from my residential IP, and the reliability on your instance is still 100%, that's neat :rocket:

**Review by Bnyro — APPROVED** [review]

**Comment by return42** [comment]
FYI: before merge, lets wait for the Docker build of PR https://github.com/searxng/searxng/pull/3938 to be successful --> https://github.com/searxng/searxng/actions/runs/11574212076/job/32218299745

**Pull request merged**